src/main/java/com/company/managementservice/controller/EmployeeController.java

```diff
@@ -0,0 +1,152 @@
+                errMsg += err.getField() + " is " + err.getCode();
+            }
+            throw new MethodArgumentNotValidException(errMsg);
+        }
+        return new ServiceResponse<>(
+                new BaseMessageResponse(employeeService.saveEmployee(employeeDto),
+                                        HttpStatus.OK, true));
+
+    }
+
+    @PutMapping(value = "/{employeeId}/assign-department/{departmentId}/in-organisation/{organisationId}")
+    public ServiceResponse<?> assignEmployeeDepartment(@NonNull @PathVariable Long employeeId,
+                                                       @NonNull @PathVariable Long departmentId,
+                                                       @NonNull @PathVariable Integer organisationId
+    )
+            throws NotFoundException, RequestRejectedException {
+        log.info(
+                "EmployeeController : assignEmployeeToDepartment : Received Request to assign Department To Employee:{} :{}"
+                , departmentId, employeeId);
+        if (employeeId == null || departmentId == null || organisationId == null)
+            throw new RequestRejectedException("Provide valid id ");
+
+        return new ServiceResponse<>(
+                new BaseMessageResponse(
+                        employeeService.putEmployeeToDepartment(employeeId, departmentId, organisationId),
+                        HttpStatus.OK, true));
+
+    }
+
+    @PutMapping(value = "/{employeeId}/freelance/organisation/{organisationId}")
+    public ServiceResponse<?> assignFreelancerEmployeeOrganisation(@NonNull @PathVariable Long employeeId,
+                                                                   @NonNull @PathVariable Integer organisationId
+    )
+            throws NotFoundException, RequestRejectedException {
+        log.info(
+                "EmployeeController : assignEmployeeToDepartment : Received Request to assign Department To Employee:{} :{}"
+                , employeeId, organisationId);
+        if (employeeId == null || organisationId == null)
+            throw new RequestRejectedException("Provide valid id ");
+
+        return new ServiceResponse<>(
+                new BaseMessageResponse(
+                        employeeService.putFreelancerEmployeeToOrganiation(employeeId, organisationId),
+                        HttpStatus.OK, true));
+
+    }
+
+    @DeleteMapping(value = "/{employeeId}/remove-from-department/{departmentId}")
+    public ServiceResponse<?> removeEmployeeFromDepartment(@NonNull @PathVariable Long departmentId,
+                                                           @NonNull @PathVariable Long employeeId
+    )
+            throws NotFoundException, RequestRejectedException {
+        log.info(
+                "EmployeeController : removeFromDepartment : Received Request to remove Employee from Department :{} :{}"
+                , employeeId, departmentId);
+        if (employeeId == null || departmentId == null)
```

Comment:
first two conditions are redundant

Reply:
spring validators should be checking nonnull, or if they are not function, they should be removed.